cmd/podman/oci-trace-hook/trace.go

```diff
@@ -0,0 +1,238 @@
+};
+
+int enter_trace(struct tracepoint__raw_syscalls__sys_enter *args)
+{
+    struct data_t data = {};
+    u64 key = 0;
+    unsigned int zero = 0;
+    struct task_struct *task;
+
+    data.pid = bpf_get_current_pid_tgid();
+    data.id = (int)args->id;
+    bpf_get_current_comm(&data.comm, sizeof(data.comm));
+
+    task = (struct task_struct *)bpf_get_current_task();
+    struct nsproxy *ns = task->nsproxy;
+    unsigned int inum = ns->pid_ns_for_children->ns.inum;
+
+    if (data.pid == PARENT_PID)
+    {
+        parent_namespace.update(&key, &inum);
+    }
+    unsigned int *parent_inum = parent_namespace.lookup_or_init(&key, &zero);
+
+    if (*parent_inum != inum)
+    {
+        return 0;
+    }
+
+    events.perf_submit(args, &data, sizeof(data));
+    return 0;
+}
+`
+
+func main() {
+	logfile, err := os.OpenFile(os.TempDir()+"/logfile", os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+	if err != nil {
+		log.Fatalf("error opening file: %v", err)
+	}
+	defer logfile.Close()
+	log.SetOutput(logfile)
+	log.Println(os.Args)
+	terminate := flag.Bool("e", false, "send SIGINT to floating process")
+	runBPF := flag.Int("r", 0, "-r <pid of the container>")
+	flag.Parse()
+
+	if *runBPF > 0 {
+		syscalls := make(calls, 303)
+		pid := *runBPF
+		src := strings.Replace(source, "PARENT_PID", strconv.Itoa(pid), -1)
+		m := bpf.NewModule(src, []string{})
+		defer m.Close()
+
+		tracepoint, err := m.LoadTracepoint("enter_trace")
+		if err != nil {
+			log.Println(err)
+		}
+
+		if err := m.AttachTracepoint("raw_syscalls:sys_enter", tracepoint); err != nil {
+			log.Println("unable to load tracepoint")
+		}
+
+		log.Println("tracepoint attached")
+
+		table := bpf.NewTable(m.TableId("events"), m)
+		channel := make(chan []byte)
+		perfMap, err := bpf.InitPerfMap(table, channel)
+		if err != nil {
+			log.Println("unable to init perf map")
+		}
+
+		sig := make(chan os.Signal, 1)
+		signal.Notify(sig, os.Interrupt, os.Kill)
+		rsc := false
+		go func() {
+			var e event
+			for {
+				data := <-channel
+				err := binary.Read(bytes.NewBuffer(data), binary.LittleEndian, &e)
+				if err != nil {
+					log.Printf("failed to decode received data '%s': %s\n", data, err)
+					continue
+				}
+				// comm := (*C.char)(unsafe.Pointer(&e.Command))
+				// log.Printf("Pid : %d, Syscall_ID : %d, , Command : %q", e.Pid, e.Id, C.GoString(comm))
+				name := getName(e.ID)
+				if name == "seccomp" {
+					rsc = true
+					continue
+				}
+				if rsc {
+					syscalls[name]++
+				}
+			}
+		}()
+		log.Println("PerfMap Start")
+		perfMap.Start()
+		<-sig
+		log.Println("PerfMap Stop")
+		perfMap.Stop()
+		generateProfile(syscalls, os.TempDir()+"/profile.json")
+	} else if *terminate {
+		log.Println("terminator")
+		f, err := ioutil.ReadFile("pid")
+
+		if err != nil {
+			log.Println(err)
+		}
+
+		log.Println("file opened")
+		Spid := string(f)
+
+		log.Println(Spid)
+		pid, _ := strconv.Atoi(Spid)
+		p, _ := os.FindProcess(pid)
+		p.Signal(os.Interrupt)
+
+	} else {
```

Comment:
What is this third option? Generating BPF and Terminating make sense, they have CLI arguments, but this one isn't documented at all

Reply:
This is still valid. Please add a new flag for that. It is confusing that there is silent default. If non of the flags is specified, we can error out.